transport/http/outbound.go

```diff
@@ -323,6 +339,13 @@ func (o *Outbound) withOpentracingSpan(ctx context.Context, req *http.Request, t
 }
 
 func (o *Outbound) withCoreHeaders(req *http.Request, treq *transport.Request, ttl time.Duration) *http.Request {
+	// Add default headers to all requests.
+	for k, vs := range o.headers {
+		for _, v := range vs {
+			req.Header.Add(k, v)
```

Comment:
We can forbid adding headers that start with "Rpc-" in AddHeader. We can loosen the restriction later if there are use cases where people want to inject conflicting headers for some reason.